firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -41,22 +41,42 @@ void moveDiagonal(int dirX, int dirY, int spacesX, int spacesY)
 
   // similary subject to change as above function dictation, same number as above though
   float numStepsX = spacesX * stepsPerSpace;
-  float numStepsX = spacesY * stepsPerSpace;
+  float numStepsY = spacesY * stepsPerSpace;
 
   digitalWrite(MOTOR_SLEEP, HIGH);
   digitalWrite(MOTOR_RESET, HIGH);
   digitalWrite(MOTOR_ENABLE, LOW);
 
   digitalWrite(xMotor[1], dirX);
-  digitalWrite(yMotor[1], dirX);
-
-  for(i = 0; i < numSteps; i++)
+  digitalWrite(yMotor[1], dirY);
+  
+  if (numStepsX = numStepsY) 
```

Comment:
This should be ==, right now I't doing an assignment and always evaluating true cuz, for arduino, >0 is considered true